<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LangChain Avanzado</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* === ESTILO COPIADO DE TU VENTANA BÁSICA === */
QDialog {
    background-color: #f6f7fb;
    color: #333333;
    font-family: "Segoe UI Variable", "Roboto", "Open Sans", sans-serif;
    font-size: 15px;
    font-weight: 300;
}

/* === Frames === */
QFrame {
    background-color: #ffffff;
    border-radius: 10px;
    border: 1px solid #dcdcdc;
}

/* Barra superior */
#frame_superior {
    background-color: #e8ecf1;
    border: none;
}

/* Panel lateral */
#frame_lateral {
    background-color: #f1f3f6;
    border-right: 1px solid #d1d1d1;
}

/* Área principal */
#frame_paginas {
    background-color: #ffffff;
}

/* === Botones Generales === */
QPushButton {
    background-color: #f5f5f5;
    color: #333333;
    border: 1px solid #cccccc;
    border-radius: 6px;
    padding: 8px 14px;
    font-weight: 400;
}

QPushButton:hover {
    background-color: #0078d7;
    color: #ffffff;
    border: 1px solid #0078d7;
}

QPushButton:pressed {
    background-color: #005a9e;
    color: #ffffff;
    border: 1px solid #005a9e;
}

/* === Botón MENÚ (Igual que en Básicos) === */
#boton_menu {
    background-color: #0078d7;
    color: white;
    border: none;
    font-weight: 400;
}
#boton_menu:hover {
    background-color: #006cbe;
}
#boton_menu:pressed {
    background-color: #005a9e;
}

/* === Botón SALIR (Igual que en Básicos) === */
#boton_salir {
    background-color: #e74c3c;
    color: white;
    border: none;
}
#boton_salir:hover {
    background-color: #c0392b;
}
#boton_salir:pressed {
    background-color: #a93226;
}

/* === Botones del Menú Lateral === */
#btn_lc_1, #btn_lc_2, #btn_lc_3, #btn_lc_4, #btn_lc_5, #btn_lc_6, #btn_lc_7, #btn_lc_8 {
    background-color: #ffffff;
    border: 1px solid #cccccc;
    color: #333333;
    text-align: left;
    padding-left: 20px;
}

#btn_lc_1:hover, #btn_lc_2:hover, #btn_lc_3:hover, #btn_lc_4:hover, #btn_lc_5:hover, #btn_lc_6:hover, #btn_lc_7:hover, #btn_lc_8:hover {
    background-color: #e8f0fe;
    border: 1px solid #0078d7;
    color: #0078d7;
}

/* === Inputs y TextEdits === */
QLineEdit, QTextEdit {
    background-color: #ffffff;
    color: #333333;
    border: 1px solid #cccccc;
    border-radius: 4px;
    padding: 6px 8px;
}

QLineEdit:focus, QTextEdit:focus {
    border: 1px solid #0078d7;
    background-color: #f9fcff;
}

/* === Labels === */
QLabel {
    color: #333333;
    border: none;
    font-weight: 300;
}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,8">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frame_superior">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="boton_menu">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Menú</string>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>../imagenes/boton_menu.png</normaloff>../imagenes/boton_menu.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>64</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Modelos LangChain</string>
        </property>
        <property name="styleSheet">
         <string notr="true">font-size: 18px; font-weight: 400;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>64</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="boton_salir">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>../imagenes/boton_cerrar.png</normaloff>../imagenes/boton_cerrar.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_central">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QFrame" name="frame_lateral">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>0</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QPushButton" name="btn_lc_1">
           <property name="text">
            <string>1. LLMChain</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_2">
           <property name="text">
            <string>2. SequentialChain</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_3">
           <property name="text">
            <string>3. SimpleSequential</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_4">
           <property name="text">
            <string>4. Parseo</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_5">
           <property name="text">
            <string>5. Varios Pasos</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_6">
           <property name="text">
            <string>6. Memoria</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_7">
           <property name="text">
            <string>7. Persistencia</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_lc_8">
           <property name="text">
            <string>8. RAG</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame_paginas">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QStackedWidget" name="stackedWidget">
           
           <widget class="QWidget" name="page_lc_1">
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QLabel" name="label_lc_1">
               <property name="text">
                <string>Página 1: LLMChain Simple (Contexto + Tema)</string>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; font-size: 16px; color: #0078d7;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_ctx_1">
               <property name="text">
                <string>1. Contexto / Rol (Ej: Actúa como un pirata):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_contexto_lc_1">
               <property name="placeholderText">
                <string>Escribe aquí el rol...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_tema_1">
               <property name="text">
                <string>2. Tema (Ej: La fotosíntesis):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_lc_1">
               <property name="placeholderText">
                <string>Escribe aquí el tema...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_enviar_lc_1">
               <property name="text">
                <string>Generar Respuesta</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QTextEdit" name="output_lc_1">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           
           <widget class="QWidget" name="page_lc_2">
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QLabel" name="label_lc_2">
               <property name="text">
                <string>Página 2: Resumen + Traducción (Secuencial)</string>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; font-size: 16px; color: #0078d7;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_instruccion_idioma">
               <property name="text">
                <string>1. Idioma de destino:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_idioma_lc_2">
               <property name="placeholderText">
                <string>Ej: Francés, Japonés, Inglés...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_instruccion_texto">
               <property name="text">
                <string>2. Texto original (Español):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_lc_2">
               <property name="placeholderText">
                <string>Pega aquí el texto largo...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_enviar_lc_2">
               <property name="text">
                <string>Ejecutar Cadena</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QTextEdit" name="output_lc_2">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>

           <widget class="QWidget" name="page_lc_3">
            <layout class="QVBoxLayout" name="verticalLayout_6">
             
             <item>
              <widget class="QLabel" name="label_lc_3">
               <property name="text">
                <string>Página 3: Cadena Libre (Tú defines los 2 pasos)</string>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; font-size: 16px; color: #0078d7;</string>
               </property>
              </widget>
             </item>

             <item>
              <widget class="QLabel" name="lbl_inst_1">
               <property name="text">
                <string>1. Primera Orden (Ej: Escribe un chiste sobre...):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_instruccion_1_lc_3">
               <property name="placeholderText">
                <string>Escribe aquí la primera instrucción...</string>
               </property>
              </widget>
             </item>

             <item>
              <widget class="QLabel" name="lbl_inst_2">
               <property name="text">
                <string>2. Segunda Orden (Ej: Explica por qué es gracioso...):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_instruccion_2_lc_3">
               <property name="placeholderText">
                <string>Escribe aquí la segunda instrucción (se aplicará al resultado anterior)...</string>
               </property>
              </widget>
             </item>

             <item>
              <widget class="QLabel" name="lbl_input_3">
               <property name="text">
                <string>3. Tema / Objeto Inicial (Ej: La programación):</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="input_lc_3">
               <property name="placeholderText">
                <string>Escribe el tema inicial...</string>
               </property>
              </widget>
             </item>

             <item>
              <widget class="QPushButton" name="btn_enviar_lc_3">
               <property name="text">
                <string>Ejecutar Secuencia</string>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; padding: 8px;</string>
               </property>
              </widget>
             </item>

             <item>
              <widget class="QTextEdit" name="output_lc_3">
               <property name="readOnly">
                <bool>true</bool>
               </property>
               <property name="placeholderText">
                <string>El resultado de la cadena aparecerá aquí...</string>
               </property>
              </widget>
             </item>
             
            </layout>
           </widget>

           <widget class="QWidget" name="page_lc_4">
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item><widget class="QLabel" name="l4"><property name="text"><string>Página 4</string></property></widget></item>
             <item><widget class="QLineEdit" name="input_lc_4"/></item>
             <item><widget class="QPushButton" name="btn_enviar_lc_4"><property name="text"><string>Enviar</string></property></widget></item>
             <item><widget class="QTextEdit" name="output_lc_4"/></item>
            </layout>
           </widget>

           <widget class="QWidget" name="page_lc_5">
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item><widget class="QLabel" name="l5"><property name="text"><string>Página 5</string></property></widget></item>
             <item><widget class="QLineEdit" name="input_lc_5"/></item>
             <item><widget class="QPushButton" name="btn_enviar_lc_5"><property name="text"><string>Enviar</string></property></widget></item>
             <item><widget class="QTextEdit" name="output_lc_5"/></item>
            </layout>
           </widget>

           <widget class="QWidget" name="page_lc_6">
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item><widget class="QLabel" name="l6"><property name="text"><string>Página 6</string></property></widget></item>
             <item><widget class="QLineEdit" name="input_lc_6"/></item>
             <item><widget class="QPushButton" name="btn_enviar_lc_6"><property name="text"><string>Enviar</string></property></widget></item>
             <item><widget class="QTextEdit" name="output_lc_6"/></item>
            </layout>
           </widget>

           <widget class="QWidget" name="page_lc_7">
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item><widget class="QLabel" name="l7"><property name="text"><string>Página 7</string></property></widget></item>
             <item><widget class="QLineEdit" name="input_lc_7"/></item>
             <item><widget class="QPushButton" name="btn_enviar_lc_7"><property name="text"><string>Enviar</string></property></widget></item>
             <item><widget class="QTextEdit" name="output_lc_7"/></item>
            </layout>
           </widget>

           <widget class="QWidget" name="page_lc_8">
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <item><widget class="QLabel" name="l8"><property name="text"><string>Página 8</string></property></widget></item>
             <item><widget class="QLineEdit" name="input_lc_8"/></item>
             <item><widget class="QPushButton" name="btn_enviar_lc_8"><property name="text"><string>Enviar</string></property></widget></item>
             <item><widget class="QTextEdit" name="output_lc_8"/></item>
            </layout>
           </widget>

          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>